### PR TITLE
[TTS] Fix TTS audio preprocessing script

### PR DESCRIPTION
**Pull request by rlangman**
# What does this PR do ?

Fix some bugs and improve usability of TTS preprocessing script.

**Collection**: [TTS]

# Changelog
- Replace hydra runner based script with regular python script that uses OmegaConf to parse .yaml config. The hydra script used AudioPreprocessingConfig as input, but in some situations this might not work because classes in the "scripts" directory are typically not installed or importable.
- Update the audio trimmer class paths because they were moved to "tts.parts.preprocessing", and moved them from "scripts" to "examples".
- Support relative paths in manifest.
- Ignore min_duration and max_duration filters when they are not provided.

# Before your PR is "Ready for review"
**Pre checks**:
- [x] Make sure you read and followed [Contributor guidelines](https://github.com/NVIDIA/NeMo/blob/main/CONTRIBUTING.md)
- [ ] Did you write any new necessary tests?
- [ ] Did you add or update any necessary documentation?
- [ ] Does the PR affect components that are optional to install? (Ex: Numba, Pynini, Apex etc)
  - [ ] Reviewer: Does the PR have correct import guards for all optional libraries?
  
**PR Type**:
- [ ] New Feature
- [x] Bugfix
- [ ] Documentation

### CODE REVIEW

File: examples/tts/conf/trim/energy.yaml
Comment (redoctopus) [inline examples/tts/conf/trim/energy.yaml:1]: Should this be a `git mv` for better tracking of the file?

Comment (rlangman) [reply examples/tts/conf/trim/energy.yaml:1]: I just modified it to be 2 commits; one which only renames with git mv and the other which updates the target path. Though the final diff on github with both commits still shows it as a deletion + creation for some reason.

**Review by redoctopus — DISMISSED** [review]
Ah well, even if it shows up as a delete and an add, not the end of the world. LGTM

File: nemo/collections/tts/parts/utils/tts_dataset_utils.py [hunk 1]
```diff
@@ -23,6 +23,17 @@
 from torch.special import gammaln
 
 
+def get_audio_paths(audio_path, base_path):
+    if os.path.isabs(audio_path):
+        abs_path = audio_path
+        rel_path = audio_path.relative_to(base_path)
```
Comment (XuesongYang) [inline nemo/collections/tts/parts/utils/tts_dataset_utils.py:29]: `relative_to` is a function for `pathlib.Path` object. Could you pls add type hints in the function?

Comment (XuesongYang) [inline nemo/collections/tts/parts/utils/tts_dataset_utils.py:29]: `relative_to` is a function for `pathlib.Path` object. Could you pls add type hints in the function?

File: scripts/dataset_processing/tts/audio_processing/preprocess_audio.py [hunk 2]
```diff
@@ -99,53 +136,51 @@ def _process_entry(
...
 
+    output_path.parent.mkdir(exist_ok=True, parents=True)
     sf.write(file=output_path, data=audio, samplerate=sample_rate)
 
-    original_duration = librosa.get_duration(filename=str(audio_filepath))
+    original_duration = librosa.get_duration(filename=str(audio_path))
     output_duration = librosa.get_duration(filename=str(output_path))
 
-    entry["audio_filepath"] = output_path
+    entry["audio_filepath"] = str(output_audio_filepath)
```
Comment (XuesongYang) [inline scripts/dataset_processing/tts/audio_processing/preprocess_audio.py:145]: nit: this function actually applied many forced conversions from `pathlib.Path` to `str`. I feel it is redundant. Maybe we can only specify a `str` type for `input_audio_dir` and `output_audio_dir`?

File: nemo/collections/tts/parts/utils/tts_dataset_utils.py (hunk 1 above)
Comment (rlangman) [reply nemo/collections/tts/parts/utils/tts_dataset_utils.py:30]: Fixed

**Review by XuesongYang — APPROVED** [review]

File: scripts/dataset_processing/tts/audio_processing/preprocess_audio.py (hunk 2 above)
Comment (XuesongYang) [reply scripts/dataset_processing/tts/audio_processing/preprocess_audio.py]: looks good to me. Thanks.

**Pull request merged**